controllers/status.go

```diff
@@ -0,0 +1,185 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// The default set of status change reasons.
+const (
+	ReasonAsExpected   = "AsExpected"
+	ReasonInitializing = "Initializing"
+	ReasonSyncing      = "SyncingResources"
+	ReasonSyncFailed   = "SyncingFailed"
+)
+
+const (
+	clusterOperatorName        = "cloud-controller-manager"
+	operatorVersionKey         = "operator"
+	defaultManagementNamespace = "openshift-cloud-controller-manager-operator"
+)
+
+// statusDegraded sets the Degraded condition to True, with the given reason and
+// message, and sets the upgradeable condition.  It does not modify any existing
+// Available or Progressing conditions.
+func (r *CloudOperatorReconciler) statusDegraded(ctx context.Context, reconcileErr error) error {
```

Comment:
suggest to add verb in this functions names, e.g `setStatusDegraded`. WDYT?

Reply:
We call it `statusDegraded` in MAO and in cluster-machine-approver. cc @JoelSpeed, do you think it is ok to rename these methods?

Reply:
I'm happy to rename so long as they are consistent within a single operator. IMO `setStatusFoo` is better than `statusFoo` anyway 😄

Reply:
Changed.